### PR TITLE
chore: Rework SvgEdit & related utils

### PR DESCRIPTION
**Pull request by evancharlton**
This change started as an attack on the strict mode violations related to `SvgEdit` (the highest number in a single file, but also the related cluster). In the process, much of the inner mechanics of the `SvgEdit` component were reworked, with a focus on compartmentalization (for maintainability / extensibility) and performance.

There is no observable feature difference in my (limited?) testing. However, I am willing to believe that there are some bugs and/or features inadvertently dropped.

This change also pulls `SvgEdit` into Storybook, so that it can be tested in isolation - including by devs who might not have permissions to create all of the necessary data to test in production. This is a pattern which I would like to repeat in future components.

### CODE REVIEW

File: src/components/SvgEdit.tsx [hunk 1]
```diff
@@ -1,707 +0,0 @@
...
-  function handleMouseUp(e) {
-    // Remove selection caused by shift-button used to create new points
-    document.getSelection().removeAllRanges();
-    e.preventDefault();
-    // Overlap with an existing path if the two last points overlap
-    if (readOnlyPoints?.length >= 1 && points.length >= 2 && activePoint > 0) {
-      const p0 = points[activePoint - 1];
-      const p1 = points[activePoint];
-      const ix = readOnlyPoints?.findIndex((p) => p1.x === p.x && p1.y === p.y);
-      if (
```
Comment (evancharlton) [inline src/components/SvgEdit.tsx:275]: @jossi87 - I couldn't find out what this is intended to do (by playing around with the deployed version). Can you explain the intended use case to me? Then I can port it to the new implementation.

Comment (jossi87) [reply src/components/SvgEdit.tsx:275]: [screen-capture.webm](https://github.com/jossi87/climbing-web/assets/17472908/08af4344-52af-4f8d-b75f-6b6f418cb203)
I think this is easier to explain with a video :)
It merges the path you are drawing with existing topo-paths on image (if the points you draw are close to an existing path). It also adds a point in the beginning if the first two points match the two first points on existing path. This is helpful while drawing topos like this one (where several lines share a start): https://brattelinjer.no/sector/2862?idMedia=20635
Is this helpful? Let me know if you have more questions. I'll leave this pull request open until I get feedback.

Comment (evancharlton) [reply src/components/SvgEdit.tsx:275]: That's very helpful - thanks! I'll get this ported now that I know what to look for.

**Comment by jossi87** [comment]
I tested it locally, and everything works as expected. Great work!

**Pull request merged**